src/lib/search.js

```diff
@@ -0,0 +1,135 @@
+var toGitUrl = require('github-url-from-git');
+var npm = require('npm');
+var devnull = require('fs').openSync('/dev/null', 'w');
+var redis = require('redis').createClient({
+    host: 'redis'
+});
+
+var BreakException = {};
+
+var Connection = require('cradle').Connection;
+var couch = new Connection('couchdb', 5984, {
+    cache: true
+});
+
+var setupComplete = false;
+
+var db = couch.database('registry');
+
+// load npm
+function initNPM(callback) {
+    npm.load({ logfd: devnull, outfd: devnull }, callback);
+}
+
+// Setup the DB
+function initDB(callback) {
+    db.exists(function (err, exists) {
+        if (err) {
+          console.error(err);
+          callback(err);
+        }
+
+        // TODO: Remove this and figure out a way to update the database while preventing duplication
+        if (exists) {
+            db.destroy();
+        }
+
+        console.log('Creating registry database');
+        db.create();
+        callback(null);
+    });
+}
+
+// Get all the mongoose packages from mongoose and load them into couchDB
+function populateDB(callback) {
+    npm.commands.search(['mongoose'], function(error, payload) {
```

Comment:
One of the big problems is that the memory overhead of this command is huge, this op by itself is enough to crash the process on an EC2 t2.micro. I'm really interested to see if there's any way we can convert this to a stream so we're not dependent on having enough memory to store all the data that npm spits out at once.